cleanup-spurious-images.py

```diff
@@ -131,21 +137,36 @@ def cluster_connect(pool, conffile, rados_id):
     ioctx = cluster_connect(cfg.pool, cfg.conf, cfg.user)
     rbd_inst = rbd.RBD()
     sess = make_session(cfg)
-    cclient = cinder_client.Client('2', session=sess)
+    cclient = cinder_client.Client('2', session=sess,region_name=cfg.os_region_name)
+
+    vols_snaps = {}
+
+    for vol in rbd_inst.list(ioctx):
+        if volume_re.match(vol):
+            image = rbd.Image(ioctx,vol)
+            vols_snaps[vol]=rbd.Image.list_snaps(image)
 
-    volumenames = [vol for vol in rbd_inst.list(ioctx) if volume_re.match(vol)]
-    log.info("Got information about %d volumes", len(volumenames))
-    # Inizializza una lista
     to_delete= []
-    for name in volumenames:
-        uuid = volume_re.search(name).group('uuid')
+    for vol in vols_snaps:
+        uuid = volume_re.search(vol).group('uuid')
```

Comment:
This assumes that `vol` matches `volume_re`, which might not be true in the future if some implementation detail of cinder and/or Ceph changes.  I'd rather make it more future-proof like e.g. this::

    match = volume_re.search(vol)
    if match:
        uuid = match.group('uuid')
    else:
        raise SomeError("Unexpected volume name: `%s`" % vol)